src/core/Akka.Cluster/Cluster.cs

```diff
@@ -221,13 +225,24 @@ public void Down(Address address)
         /// Typically used together with configuration option 'akka.cluster.min-nr-of-members' to defer some action,
         /// such as starting actors, until the cluster has reached a certain size.
         /// </summary>
-        /// <param name="callback"></param>
+        /// <param name="callback">The callback that will be run whenever the current member achieves a status of <see cref="MemberStatus.Up"/></param>
         public void RegisterOnMemberUp(Action callback)
         {
             _clusterDaemons.Tell(new InternalClusterAction.AddOnMemberUpListener(callback));
         }
 
         /// <summary>
+        /// The supplied callback will be run once when the current cluster member is <see cref="MemberStatus.Removed"/>.
+        /// 
+        /// Typically used in combination with <see cref="Leave"/> and <see cref="ActorSystem.Terminate"/>.
+        /// </summary>
+        /// <param name="callback">The callback that will be run whenever the current member achieves a status of <see cref="MemberStatus.Down"/></param>
+        public void RegisterOnMemberRemoved(Action callback)
+        {
+            _clusterDaemons.Tell(new InternalClusterAction.AddOnMemberRemovedListener(callback));
```

Comment:
Ah, nice catch - otherwise the hook would never fire if the cluster was already terminated.